scenarios/two_party_countersigning/src/main.rs

```diff
@@ -117,132 +119,109 @@ fn agent_behaviour_initiate(
-
-                                    let elapsed = start.elapsed();
-
-                                    log::debug!("Completed the countersigning session with agent {:?}", agent_pub_key);
-
-                                    let initiated_success = initiated_success.fetch_add(1, std::sync::atomic::Ordering::Acquire);
-                                    reporter.add_custom(
-                                        ReportMetric::new("countersigning_session_initiated_success")
-                                            .with_tag("agent", agent_name.clone())
-                                            .with_field("retries", retry_count as u64)
-                                            .with_field("value", initiated_success as u64),
-                                    );
-                                    reporter.add_custom(
-                                        ReportMetric::new("countersigning_session_initiated_duration")
-                                            .with_tag("agent", agent_name)
-                                            .with_field("value", elapsed.as_secs_f64()),
-                                    );
-
-                                    break;
-                                }
-                                None => {
-                                    log::warn!("No signal received, problem with the remote? Will try again.");
-                                }
-                            }
-                        }
+                    let session_times = my_preflight_response.request.session_times.clone();
+                    let session_timeout = Instant::now().add(Duration::from_millis(
+                        (session_times.end.as_millis() - session_times.start.as_millis()) as u64,
+                    ));
 
-                        // Add no new agents, that should only happen when we exhaust the list.
-                        Ok(Vec::with_capacity(0))
+                    match run_initiated_session(
+                        client.clone(),
+                        my_preflight_response,
+                        session_timeout,
+                        agent_pub_key,
+                        agent_name.clone(),
+                        cell_id.clone(),
+                        app_port,
+                        start,
+                        initiated_success.clone(),
+                        reporter,
+                    )
+                    .await
+                    {
+                        Ok(_) => {
+                            // Completed successfully
+                        }
+                        Err(e) => {
+                            log::warn!(
+                                "Failed to run initiated session, waiting for it to time out: {:?}",
+                                e
+                            );
+
+                            // TODO this can be replaced with listening for the abandoned signal
```

Comment:
- [ ] Timeout signals need to be handled in a couple of places through this test. Update this once Holochain is released with that signal defined